### ISSUE TITLE
Configurable rights to create/attach convoy storage volumes [feature]

### ISSUE THREAD

**Issue by sshipway** [rancher/rancher#3403]
It would be helpful to be able to control which group of users have the rights to create new storage volumes, or attach to existing ones.  Currently, anyone who can spin up a container can connect to convoy and create a new volume or mount an existing one.  It would be useful to be able to restrict this in some way.

Also, storage volumes should be able to be flagged as 'shareable' or 'exclusive', so that if required, a volume can be blocked from being mounted by a container if it is already mounted by a different container, preventing accidental data corruption

**Comment by will-chan** [comment]
We no longer support ConvoyNFS.  Use RancherNFS.
